pdr_backend/lake/table.py

```diff
@@ -2,50 +2,42 @@
 import os
 from typing import Dict, Callable
 import polars as pl
+from polars.type_aliases import SchemaDict
+
 from enforce_typing import enforce_types
 from pdr_backend.ppss.ppss import PPSS
 from pdr_backend.lake.plutil import has_data, newest_ut
 from pdr_backend.util.networkutil import get_sapphire_postfix
 from pdr_backend.util.time_types import UnixTimeMs
 from pdr_backend.lake.plutil import _object_list_to_df
 from pdr_backend.lake.table_pdr_predictions import _transform_timestamp_to_ms
+from pdr_backend.lake.csv_data_store import CSVDataStore
 
 logger = logging.getLogger("table")
 
 
 @enforce_types
 class Table:
-    def __init__(self, table_name: str, df_schema: object, ppss: PPSS):
+    def __init__(self, table_name: str, df_schema: SchemaDict, ppss: PPSS):
         self.ppss = ppss
         self.table_name = table_name
         self.df_schema = df_schema
-        self.df = pl.DataFrame()
+        self.df = pl.DataFrame([], schema=df_schema)
+        print("self.df", self.df)
         self.load()
 
     @enforce_types
     def load(self):
         """
         Read the data from the Parquet file into a DataFrame object
         """
-        filename = self._parquet_filename()
+        print(f"Loading data for {self.table_name}")
+        self.csv_data_store = CSVDataStore(self.ppss.lake_ss.parquet_dir)
         st_ut = self.ppss.lake_ss.st_timestamp
         fin_ut = self.ppss.lake_ss.fin_timestamp
-
-        # load all data from file
-        # check if file exists
-        # if file doesn't exist, return an empty dataframe with the expected schema
-        if os.path.exists(filename):
-            logger.info("Loading parquet for %s", self.table_name)
-            df = pl.read_parquet(filename)
-            df = df.filter(
-                (pl.col("timestamp") >= st_ut) & (pl.col("timestamp") <= fin_ut)
-            )
-        else:
-            logger.info("Create initial df for %s", self.table_name)
-            df = pl.DataFrame(schema=self.df_schema)
-
-        # save data frame in memory
-        self.df = df
+        self.df = self.csv_data_store.read(
+            self.table_name, st_ut, fin_ut, schema=self.df_schema
+        )
```

Comment:
This function used to do a polars query/return a dataframe from a parquet file.
This needs to be updated to do a SQL query from a DuckDB PersistentStore

Reading from CSV like this should be used for loading data into DuckDB..

Reply:
There are now two `load_into_cache` functions.
- one for csv
- one for duckdb persistent store

Resolving conversation